src/components/vm/consoles/vncAddEdit.jsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright 2024 Fsas Technologies Inc.
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    Form, Modal, ModalVariant,
+    FormGroup, FormHelperText, HelperText, HelperTextItem,
+    InputGroup, TextInput, Button,
+} from "@patternfly/react-core";
+import { EyeIcon, EyeSlashIcon } from "@patternfly/react-icons";
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { DialogsContext } from 'dialogs.jsx';
+import { domainChangeVncSettings, domainAttachVnc, domainGet } from '../../../libvirtApi/domain.js';
+import { NeedsShutdownAlert } from '../../common/needsShutdown.jsx';
+
+const _ = cockpit.gettext;
+
+const VncBody = ({ idPrefix, onValueChanged, dialogValues, validationErrors }) => {
+    const [showPassword, setShowPassword] = useState(false);
+
+    return (
+        <>
+            <FormGroup
+                fieldId={`${idPrefix}-portmode`}
+                label={_("Port")} isInline hasNoPaddingTop isStack>
+                <TextInput
+                    id={`${idPrefix}-port`}
+                    value={dialogValues.vncPort}
+                    type="text"
+                    validated={validationErrors?.vncPort ? "error" : null}
+                    onChange={(event) => onValueChanged('vncPort', event.target.value)} />
+                <FormHelperText>
+                    <HelperText>
+                        { validationErrors?.vncPort
+                            ? <HelperTextItem variant='error'>{validationErrors?.vncPort}</HelperTextItem>
+                            : <HelperTextItem>
+                                {_("Leave empty to automatically assign a free port when machine starts")}
```

Comment:
I previously suggested some grayed out "automatic" hint text, but I buy that this is against the PF rules. This is a nice solution, thanks! However, I think it should be "when *the* machine starts".

Reply:
Yes. Also @garrett should have the last word on this.

Reply:
Thanks, works for me now. I suggest Garrett does the final review on the screenshots.